### DISCUSSION TITLE
MFM-Doktor

### DISCUSSION THREAD

**Discussion by rchastain** [mse-org/mseide-msegui#50] [General]
Hello friends!

I had two hours to kill this morning, and I wanted to wake up my Lua knowledge, so I made this.

This is a tool made to repair MFM files that do not load correctly in MSEide, due to invalid values for properties.

Example:
```
[roland@localhost mfmdoktor]$ lua mfmdok.lua sample.mfm result.txt
MFM-Doktor 0.0.1 (Lua 5.2)
DEBUG Valid value: wo_noframe
DEBUG Valid value: wo_groupleader
DEBUG Valid value: wo_notaskbar
DEBUG Invalid value: wo_alwaysontop
DEBUG Delete: [wo_alwaysontop, ]
DEBUG Invalid value: wo_rounded
DEBUG Delete: [, wo_rounded]
DEBUG Valid value: ow_mousefocus
DEBUG Valid value: ow_tabfocus
DEBUG Valid value: ow_arrowfocus
DEBUG Valid value: ow_mousewheel
DEBUG Valid value: ow_destroywidgets
DEBUG Valid value: ow1_fontglyphheight
DEBUG Valid value: ow1_autoscale
DEBUG Valid value: ow1_autoheight
[roland@localhost mfmdoktor]$ 
```
I see in another discussion that _wo_rounded_ seems to be a valid value. I don't know why I didn't find it in my documentation.

Too tired to change this now. Suggestions, improvements, ideas are welcome.

Best regards.

Roland

[mfmdoktor-001.zip](https://github.com/mse-org/mseide-msegui/files/10247515/mfmdoktor-001.zip)

**Comment by fredvs** [comment]
> This is a tool made to repair MFM files 

Genial!
Many thanks for this precious addition.
Please add it in /mseide-msegui/tools/

> I see in another discussion that wo_rounded seems to be a valid value. I don't know why I didn't find it in my documentation.

This was a "recent" (6 months ago) addition so maybe not yet implemented when you created the doc...

**Comment by rchastain** [comment]
> Genial! Many thanks for this precious addition. Please add it in /mseide-msegui/tools/

Thanks for your support. Not sure that it deserve in its current state to be added in tools. I should update the options list, and also maybe find a more serious name. :)

But since you liked it, I will share here or in another discussion my best bash or Lua scripts. ;)

> This was a "recent" (6 months ago) addition so maybe not yet implemented when you created the doc...

OK, I see.

P.-S. I close the issue. I should have created a discussion instead, but I have just discovered that there is a "discussion" section.

**Comment by rchastain** [comment]
There was a bug in the first version of the script. (In some cases it removed valid values.)

Here is the corrected version.

[mfmdoktor-002.zip](https://github.com/mse-org/mseide-msegui/files/14482841/mfmdoktor-002.zip)

**Reply by rchastain** [reply]
I forgot to add **wo_rounded** in valid values, but it's very easy to do.

**Comment by rchastain2** [comment]
Here is MFM-Doktor 0.0.3. Now it controls also values of [coloptionty](https://msegui.net/doc/fpdoc/msegrids/coloptionty.html) type.

[mfmdok-003.zip](https://github.com/user-attachments/files/25659821/mfmdok-003.zip)
